### ISSUE TITLE
Dynamo sender fails with object that has multiple nested dictionaries

### ISSUE THREAD

**Issue by ggooskens** [speckleworks/SpeckleDynamo#55]
Step 0: 
- [x] I've  read the [contribution guidelines](https://github.com/speckleworks/SpeckleOrg/blob/master/CONTRIBUTING.md)!
#### Expected Behaviour
To organize the user data per object I was expecting it to work like in GH where you can nest multiple dictionaries per parent like here:
![ghdictionaries](https://user-images.githubusercontent.com/36707790/51535107-d6fe1d80-1e47-11e9-89d8-a0eb5571a006.PNG)

#### Actual Behaviour
Not sure if this is expected behavior. It seems only possible to nest one dictionary per parent (first stream). List of dictionaries (second stream) is also not accepted, but this is expected. Was expecting to be able to nest multiple dictionaries (third stream)
![dynamodictionaries](https://user-images.githubusercontent.com/36707790/51535489-e6319b00-1e48-11e9-82ae-988196a02870.PNG)

**Comment by alvpickmans** [comment]
What is the error thrown?

**Comment by didimitrie** [comment]
Two notes, as i just popped by this: 
- https://github.com/speckleworks/SpeckleRhino/issues/233 & https://github.com/speckleworks/SpeckleRhino/issues/183 - throw an error is key names contain `type` or `.` - life is hard, people should make do without them...
- depending on how the uds are set, you might need to call generate hashes on the object again to ensure the hash changes

**Comment by ggooskens** [comment]
> What is the error thrown?

 `"Error: One or more errors occured."` 😸

**Comment by teocomi** [comment]
@ggooskens is there a sample definition you could share? I tried with a simple nested dictionary and it seems to work:

![image](https://user-images.githubusercontent.com/2679513/51603801-1b9ebd00-1f0b-11e9-880f-92ce06007aff.png)

**Comment by ggooskens** [comment]
Sorry for the confusion, UserData.Set can indeed handle multiple nested dictionaries.
It's the sender that fails. See the example definition below (need at least one wall modelled in the active document). 
It does stream when I flatten the data (remove nestings), so key naming is probably not the issue.
![dynamodictionaries2](https://user-images.githubusercontent.com/36707790/51668173-e6a17180-1fc1-11e9-8ed1-69a7a84a226b.PNG)

Dynamo Definition:
[SendingNestedDictionaries.zip](https://github.com/speckleworks/SpeckleDynamo/files/2790966/SendingNestedDictionaries.zip)

**Comment by didimitrie** [comment]
As a reference, currently the speckle converters do not know how to handle array properties in dictionaries. It might be because of this; am looking into it but quick question, when running the script i still get out `INVALID dictionary` as the first one; it might be this that's causing the issue...

![image](https://user-images.githubusercontent.com/7696515/53095838-aaf0bd80-3515-11e9-8fe8-89db2855b6ec.png)

**Comment by ggooskens** [comment]
Hey thanks for helping! I just checked and `INVALID` is the actual value of the parametergroup of these parameters (they are not normal parameters, but store another element). I filtered them out, but still the sender fails. The problem is therefore probably indeed the list/array properties within a dictionary. Tbh, it's not a major issue at the moment :) maybe I am just doing something wrong building the dictionaries.
![dynamodictionaries3](https://user-images.githubusercontent.com/36707790/53107633-b48a1d80-3535-11e9-92e8-70eee63e08e9.PNG)
![dynamodictionaries4](https://user-images.githubusercontent.com/36707790/53107635-b48a1d80-3535-11e9-8cd1-d714b7736d97.PNG)
